### PR TITLE
 EXSWCPHIPT-132 - [catch2] Testing for hipMemsetAsync API  

### PR DESCRIPTION
**Pull request by ShanoToni**
These test cases test for multiple Asynchronous memset operations working on the same memory.
The PR is stacked on top of #2750 , which in turn is stacked on top of #2718

### CODE REVIEW

File: tests/catch/unit/memory/MemUtils.hh [hunk 1]
```diff
@@ -45,3 +45,422 @@ static void launchLongRunningKernel(clock_t milliseconds, hipStream_t stream = n
...
+
+// helper struct containing vars needed for 2D and 3D memset Testing
+struct MultiDData {
+  size_t width{};  // in elements not bytes
+  // set to 0 for 1D
+  size_t height{};  // in elements not bytes
+  // set to 0 for 2D
+  size_t depth{};   // in elements not bytes
+  size_t pitch{};   // pitch = (width * sizeofData) + alignment
+  size_t offset{};  // for simplicity use same offset for x,y and z dimentions of memory
```
Comment (FMarno) [inline tests/catch/unit/memory/MemUtils.hh]: dimensions

File: tests/catch/unit/memory/MemUtils.hh [hunk 2]
```diff
@@ -45,3 +45,422 @@ static void launchLongRunningKernel(clock_t milliseconds, hipStream_t stream = n
...
+
+template <typename T>
+static inline void devRegisteredCopy(memSetType memType, T* aPtr, T* hostMem, size_t dataW,
+                                     size_t dataH, size_t dataD, size_t& dataPitch) {
+  size_t elementSize = sizeof(T);
+
+  switch (memType) {
+    case memSetType::hipMemset3D: {
+      hipMemcpy3DParms params{};
+      params.kind = hipMemcpyHostToHost;
```
Comment (FMarno) [inline tests/catch/unit/memory/MemUtils.hh]: this is hipMemcpyHostToHost while the other hip API in this function are used with hipMemcpyDeviceToHost. Is this correct?

File: tests/catch/unit/memory/CMakeLists.txt [hunk 3]
```diff
@@ -88,6 +88,10 @@ set(TEST_SRC
     hipMallocMngdMultiThread.cc
     hipMemPrefetchAsync.cc
     hipArray.cc
+    hipFree.cc
+    hipArray.cc
```
Comment (FMarno) [inline tests/catch/unit/memory/CMakeLists.txt]: hipArray is already listed (line 90)

**Review by FMarno — DISMISSED** [review]
1 small comment, otherwise LGTM

**Review by FMarno — DISMISSED** [review]
LGTM

**Review by chrispaquot — DISMISSED** [review]

**Review by chrispaquot — DISMISSED** [review]

**Comment by mangupta** [comment]
Closing since changes are a part of #2750

**Pull request closed without merging**